middleman-core/lib/middleman-core/sitemap/store.rb

```diff
@@ -244,6 +256,20 @@ def ensure_resource_list_updated!
                 @resources.each do |resource|
                   @_lookup_by_destination_path[resource.destination_path] = resource
                 end
+               
+                if @app.config[:traversal_use_any_index]
```

Comment:
I believe `@_lookup_indexes_at_path` should be populated even when this condition evaluates to false.

For now, the `find_index_resources_by_prefix` function will not provide the appropriate index resource in case the `:traversal_use_any_index` configuration option is false because of this. Obviously that function can work around that dictionary not being populated. However I believe it would be easier (and not much more expensive) to just make sure it is always populated.

In case it is decided to have this dictionary always populated, the test condition line 267 needs to read the `:traversal_use_any_index` and the `:index_file`.

In case it is decided to have this dictionary only populated when `:traversal_use_any_index` is true, I believe this limitation needs to be clearly documented.

Reply:
Same answer as above, but I do agree around the surprising behavior. I'll mark `find_index_resources_by_prefix` as a private method so others don't rely on it.

Reply:
Fair enough, that makes sense. I believe I would not be as cautious myself, but I definitely understand.

Beware about your future self though: make sure you document this behavior so you’re not surprised when calling `find_index_resources_by_prefix` for some new use case a few years from now.

Reply:
Good idea. I'll have it throw if it wouldn't work